src/libponyc/type/lookup.c

```diff
@@ -237,13 +273,19 @@ static ast_t* lookup_union(pass_opt_t* opt, ast_t* from, ast_t* type,
             {
               // Use the supertype function. Require the most specific
               // arguments and return the least specific result.
-              // TODO: union the signatures, to handle arg names and
-              // default arguments.
-              if(errors)
-                errorframe_discard(pframe);
+              if(!param_names_match(from, result, r, name, pframe, opt))
+              {
+                ast_free_unattached(r);
+                ok = false;
+              }
```

Comment:
our `else` block handling is wildly inconsistent throughout the codebase. 

we should put together a c coding style